src/query/storage/promremote/query_converter_test.go

```diff
@@ -0,0 +1,80 @@
+package promremote
+
+import (
+	"testing"
+	"time"
+
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/m3db/m3/src/query/models"
+	"github.com/m3db/m3/src/query/storage"
+	"github.com/m3db/m3/src/query/ts"
+	xtime "github.com/m3db/m3/src/x/time"
+)
+
+func TestQueryConverter(t *testing.T) {
+	now := xtime.Now()
+	tcs := []struct {
+		name     string
+		input    storage.WriteQueryOptions
+		expected *prompb.WriteRequest
+	}{
+		{
+			name: "converts",
+			input: storage.WriteQueryOptions{
+				Tags: models.Tags{
+					Opts: models.NewTagOptions(),
+					Tags: []models.Tag{{
+						Name:  []byte("test_tag_name"),
+						Value: []byte("test_tag_value"),
+					}},
+				},
+				Datapoints: ts.Datapoints{{
+					Timestamp: now,
+					Value:     42,
+				}},
+				Unit: xtime.Millisecond,
+			},
+			expected: &prompb.WriteRequest{
+				Timeseries: []prompb.TimeSeries{
+					{
+						Labels: []prompb.Label{
+							{
+								Name:  "test_tag_name",
+								Value: "test_tag_value",
+							},
+						},
+						Samples: []prompb.Sample{
+							{
+								Timestamp: now.ToNormalizedTime(time.Millisecond),
+								Value:     42,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range tcs {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
```

Comment:
You only have one test case. What's the point of test table?

Reply:
added more cases but still removed table